baseapp/baseapp.go

```diff
@@ -534,10 +516,32 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte, tx sdk.Tx) (result sdk
 		ctx = ctx.WithMultiStore(msCache)
 	}
 
-	result = handler(ctx, msg)
+	// No need to run Msg checks when running CheckTx(), since CheckTx() only needs to validate that fees can be paid
+	// Then when DeliverTx() is called, this will run, preventing us from doing the same checks twice
+	if mode != runTxModeCheck {
+		var msg = tx.GetMsg()
+		if msg == nil {
+			return sdk.ErrInternal("Tx.GetMsg() returned nil").Result()
+		}
+
+		// Validate the Msg.
+		err := msg.ValidateBasic()
+		if err != nil {
+			err = err.WithDefaultCodespace(sdk.CodespaceRoot)
+			return err.Result()
+		}
```

Comment:
let's remove use of Validate Basic - no longer required, basic checks should just be performed at the beginning of the message handler during deliver

Reply:
okay i can delete this out of baseapp.go, but in https://github.com/cosmos/cosmos-sdk/issues/1120 it was stated that ValidateBasic() should also be removed from the message struct. Would you like me to do this as well?

Reply:
Nevermind! let's keep it in here, consensus is that there should be a self contained check on the message type still